mongo/errors.go

```diff
@@ -250,7 +249,8 @@ func (e CommandError) HasErrorCodeWithMessage(code int, message string) bool {
 
 // IsMaxTimeMSExpiredError returns true if the error is a MaxTimeMSExpired error.
 func (e CommandError) IsMaxTimeMSExpiredError() bool {
-	return e.Code == 50 || e.Name == "MaxTimeMSExpired"
+	noCode := e.Code == 0
```

Comment:
Have you considered `hasCode` (or another typical bool identifier) as the variable name instead of `noCode`?

Reply:
Thanks for your feedback @LBWright ! `hasCode` would require double negation (`hasCode := e.Code != 0; return !hasCode || ...`) which seems a little less readable. I've used `hasNoCode` instead to use a more bool-like name.